code/modules/atmospherics/machinery/components/unary_devices/bluespace_sender.dm

```diff
@@ -0,0 +1,180 @@
+/obj/machinery/atmospherics/components/unary/bluespace_sender
+	icon = 'icons/obj/atmospherics/components/bluespace_gas_selling.dmi'
+	icon_state = "bluespace_sender_off"
+	name = "Bluespace Gas Sender"
+	desc = "Sends gases to the bluespace network to be shared with the connected vendors, who knows what's beyond!"
+
+	density = TRUE
+	max_integrity = 300
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, RAD = 100, FIRE = 80, ACID = 30)
+	layer = OBJ_LAYER
+	circuit = /obj/item/circuitboard/machine/bluespace_sender
+	pipe_flags = PIPING_ONE_PER_TURF | PIPING_DEFAULT_LAYER_ONLY
+
+	///Base icon name for updating the appearance
+	var/base_icon = "bluespace_sender"
+	///Gas mixture containing the inserted gases and that is connected to the vendors
+	var/datum/gas_mixture/bluespace_network
+	///Rate of gas transfer inside the network (from 0 to 1)
+	var/gas_transfer_rate = 0.5
+	///A base price for each and every gases, in case you don't want to change them
+	var/list/base_prices = list()
+	///List storing all the vendors connected to the machine
+	var/list/vendors
+	///Amount of credits gained from each vendor
+	var/credits_gained = 0
+
+/obj/machinery/atmospherics/components/unary/bluespace_sender/Initialize()
+	. = ..()
+	initialize_directions = dir
+	bluespace_network = new
+	for(var/gas_id in GLOB.meta_gas_info)
+		bluespace_network.assert_gas(gas_id)
+	for(var/gas_id in GLOB.meta_gas_info)
+		var/datum/gas/gas = gas_id
+		base_prices |= gas_id
+		base_prices[gas_id] = initial(gas.base_value)
```

Comment:
If the entry doesn't exist already when doing `base_prices[gas_id] = initial(gas.base_value)` it will be safely made. You don't need to insert it first and you shouldn't be using `|=` anyway unless you really do need to keep uniqueness in a list and you aren't sure what has already been added. In this case where the list is starting from scratch you know it'll be unique for every one.
```suggestion
		base_prices[gas_id] = initial(gas.base_value)
```

Reply:
ah didn't know thanks